src/templates/SiteDocTemplate.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import PropsTable from 'terra-props-table';
+import Markdown from 'terra-markdown';
+import IndexExampleTemplate from '../IndexPageTemplate';
+
+const propTypes = {
+  /**
+   * The given component's current version
+   */
+  version: PropTypes.string,
+  /**
+   * The given component's readme file imported to a string
+   */
+  readme: PropTypes.string,
+  /**
+   * Holds objects with the following properties:
+   * ```
+   * title: The title of the example
+   * description: A description of the example to be displayed below the title
+   * example: The example to be displayed
+   * source: The source code of the example
+   * ```
+   */
+  examples: PropTypes.arrayOf(
+    PropTypes.shape(
+      {
+        title: PropTypes.string,
+        description: PropTypes.string,
+        example: PropTypes.element,
+        source: PropTypes.string,
+      },
+    ),
+  ),
+  /**
+   * Holds objects with the following properties:
+   * ```
+   * componentSrc: The source code of the component
+   * componentName: The name of the component
+   * ```
+   */
+  propsTables: PropTypes.arrayOf(
+    PropTypes.shape(
+      {
+        componentSrc: PropTypes.string,
+        componentName: PropTypes.string,
+      },
+    ),
+  ),
+};
+
+const defaultProps = {
+  version: '',
+  readme: '',
+  examples: [],
+  propsTables: [],
+};
+
+const SiteDocTemplate = (props) => {
+  const { version, readme, examples, propsTables } = props;
+  let id = 0;
+
+  // Assign unique identifiers to use as keys later
+  for (let i = 0; i < examples.length; i += 1) {
+    examples[i].id = id;
+    id += 1;
+  }
+
+  for (let i = 0; i < propsTables.length; i += 1) {
+    propsTables[i].id = id;
+    id += 1;
+  }
+
+  return (
+    <div>
+      {version ? <div id="version">Version: {version}</div> : null}
```

Comment:
I am wondering if there would be an clean way to append the NPM shield to the readme source to have render the version like:
<img width="99" alt="screen shot 2018-03-30 at 10 27 42 am" src="https://user-images.githubusercontent.com/14099737/38143192-01842ab4-3405-11e8-9c3e-1a8fd90ae6f2.png">

The markdown syntax for this is:
```
[![NPM version](http://img.shields.io/npm/v/terra-i18n.svg)](https://www.npmjs.org/package/terra-i18n)
```
we should need to some something like
```
// learn the package name
packageName = props.packageName;
const version = `[![NPM version](http://img.shields.io/npm/v/${packageName}.svg)](https://www.npmjs.org/package/${packageName})`

 {readme ? <Markdown src={`${version}\n${readme}`} /> : null}
```

Would anyone be opposed to dropping the `version` prop in exchange for the prop `packageName`?